dynatrace/dynatrace_test.go

```diff
@@ -188,230 +143,405 @@ func TestExporter_send(t *testing.T) {
 	})
 }
 
-type checkpointSet struct {
-	sync.RWMutex
-	records []metricsdk.Record
+func TestExporter_TemporalityFor(t *testing.T) {
+	e := &Exporter{}
+	if temporality := e.TemporalityFor(&sdkapi.Descriptor{}, aggregation.HistogramKind); temporality != aggregation.DeltaTemporality {
+		t.Errorf("Should return delta temporality for histogram - got %v", temporality.String())
+	}
+
+	counterDescriptor := sdkapi.NewDescriptor("", sdkapi.CounterInstrumentKind, number.Float64Kind, "", unit.Bytes)
+	if temporality := e.TemporalityFor(&counterDescriptor, aggregation.SumKind); temporality != aggregation.DeltaTemporality {
+		t.Errorf("Should return delta temporality for monotonic counter - got %v", temporality.String())
+	}
+
+	asyncCounterDescriptor := sdkapi.NewDescriptor("", sdkapi.CounterInstrumentKind, number.Float64Kind, "", unit.Bytes)
```

Comment:
Isnt this the exact same as above? There is no way to distinguish between sync and async at this point right?

Reply:
Should have been CounterObserverInstrumentKind